src/proteomes/components/entry/ComponentsButtons.tsx

```diff
@@ -40,6 +46,19 @@ const ComponentsButtons = ({
 }: Props) => {
   const [displayDownloadPanel, setDisplayDownloadPanel] = useState(false);
 
+  const handleToggleDownload = useCallback(
+    (reason: DownloadPanelFormCloseReason, downloadMethod?: DownloadMethod) => {
+      if (displayDownloadPanel) {
+        sendGtagEventPanelResultsDownloadClose(reason, downloadMethod);
+        setDisplayDownloadPanel(false);
+      } else {
+        sendGtagEventPanelOpen('results_download');
+        setDisplayDownloadPanel(true);
+      }
```

Comment:
```suggestion
      setDisplayDownloadPanel(displayDownloadPanel => {
        if (displayDownloadPanel) {
          sendGtagEventPanelResultsDownloadClose(reason, downloadMethod);
        } else {
          sendGtagEventPanelOpen('results_download');
        }
        setDisplayDownloadPanel(!displayDownloadPanel)
      });
```

I would do something like this, to avoid having `displayDownloadPanel` as a dependency of the effect.
This has been edited in GitHub, so if you go for something like this make sure that lint and type checks still pass